src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1847,11 +1799,11 @@ namespace winrt::TerminalApp::implementation
         }
 
         // If the user set a custom name, save it
-        if (_WindowName != L"")
+        if (const auto windowName = _WindowProperties.WindowName(); !windowName.empty())
```

Comment:
I believe that `hstring`s evaluate as booleans in this context; you can do `if(const auto windowName = ...)` alone

Reply:
_they don't_ 😭